include/sentry.h

```diff
@@ -1362,6 +1362,14 @@ SENTRY_API sentry_user_consent_t sentry_user_consent_get(void);
  */
 SENTRY_API sentry_uuid_t sentry_capture_event(sentry_value_t event);
 
+struct sentry_path_s;
+typedef struct sentry_path_s sentry_path_t;
```

Comment:
If we must expose this (can/should we just use string instead?), users would also need a way to create this opaque struct.

Reply:
Its more an issue of lack of API knowledge. I found ```sentry__path_from_str``` so going to pass in ```const char*``` and convert within the function.

Reply:
Yeah, we don't currently expose the path and we should take it as a string here. However, we do provide `_n` interfaces that take ptr/len strings in addition to null-terminated strings (because people increasingly use `strings_views` and other things that cannot guarantee null-termination), where typically the null-terminated version is implemented in terms of the ptr/len interface.

Reply:
I've added the _n variant